### ISSUE TITLE
[Feature Request] Implementation of L-BFGS optimizer

### ISSUE THREAD

**Issue by jasonyu1996** [apache/incubator-mxnet#12132]
Hi!
Would the developers consider adding an implementation of limited memory BFGS to the library? PyTorch offers one here: https://pytorch.org/docs/stable/optim.html?#torch.optim.LBFGS

**Comment by adaaaaaa** [comment]
follow with interest...

**Comment by lanking520** [comment]
@jasonyu1996 thanks for your question and  keep watching this thread.
@eric-haibin-lin do we have something similar in MXNet?

@mxnet-label-bot please label this as [feature, memory]

**Comment by anirudhacharya** [comment]
I dont think MXNet has this optimizer, and optimizer requests are tracked here - https://github.com/apache/incubator-mxnet/issues/9182

**Comment by jasonyu1996** [comment]
Thanks! I have added it to the optimizer request list. Actually I am quite willing to see if I could implement one for MXNet myself.

**Comment by haojin2** [comment]
@jasonyu1996 You're more than welcome to add this thing yourself if you want to, most of the optimizers should be here: https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/optimizer.py. You can use any of them (for example, SGD) as reference to implement this. And all unit tests for optimizers are here: https://github.com/apache/incubator-mxnet/blob/master/tests/python/unittest/test_optimizer.py. Please make sure to add a test for your new optimizer once you finish implementing it.
